x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/hooks/use_index_data.ts

```diff
@@ -105,11 +108,15 @@ export const useIndexData = (
 
   const fetchColumnChartsData = async function () {
     try {
-      const columnChartsData = await fetchChartsData(
-        indexPattern.title,
-        ml.esSearch,
-        query,
-        columns.filter((cT) => dataGrid.visibleColumns.includes(cT.id))
+      const dataLoader = new DataLoader(indexPattern, toastNotifications);
```

Comment:
I presume you shouldn't initialize  data loader on each fetch

Reply:
Changed in 3883477.